### DISCUSSION TITLE
Support v6.x

### DISCUSSION THREAD

**Discussion by vicasas** [nolimits4web/swiper#5021] [Q&A]
I was eager to be able to use Swiper v7 until I saw that it is only supported by ESM.  💣

I am left with the following questions: 

1. Do you plan to continue supporting previous versions?

2. If so, where can the v6 documentation be found?

3. Do you plan to make improvements from v7 to v6 regarding loading modules without using .use()?  To be able to programmatically control the load of the modules?

4. It was suggested that perhaps leaving CommonJS out would make many projects that use Swiper stop using them because they did not want to migrate to ESM yet?

**Comment by nolimits4web** [comment]
1. No
2. https://github.com/nolimits4web/swiper-website/tree/v6
3. No

**Comment by nolimits4web** [comment]
4. You don't have to fully migrate to ESM, for example you can:

- use bundler that just supports ESM (web pack 5, Vite, Rollup)
- transpile it to CJS by properly configuring Babel
- copy it from node_modules to your local src folders
- use UMD version

**Reply by vicasas** [reply]
@nolimits4web You could provide documentation or examples to migrate to v7 without having to use ESM.

In the projects that we have in production that use Swiper, we are not willing to have to use ESM or add additional configurations to the web packer or babel. This is why we are now looking at other solutions just because of the void v6 support and the "obligation" to use v7 with ESM.

Will it be possible to work in a commonJS version or, as I mentioned at the beginning, provide usage documentation without having to go through ESM?

**Reply by vicasas** [reply]
@nolimits4web Hi!, could you provide examples?
